packages/sync-server/src/sync/CentralSyncManager.ts

```diff
@@ -437,16 +450,121 @@ export class CentralSyncManager {
     }
   }
 
+  async persistIncomingChanges(sessionId: string, deviceId: string) {
+    const totalPushed = await countSyncSnapshotRecords(
+      this.models.database,
+      sessionId,
+      SYNC_SESSION_DIRECTION.INCOMING,
+    );
+    await this.models.syncSession.addInfo(sessionId, {
+      beganPersistAt: new Date(),
+      totalPushed,
+    });
+
+    try {
+      // commit the changes to the db
+      const persistedAtSyncTick = await this.models.wrapInTransaction(
+        async (transactingModels: SyncServerModelRegistry) => {
+          const modelsToInclude = getModelsForPush(transactingModels.getModels());
+
+          // we tick-tock the global clock to make sure there is a unique tick for these changes
+          // n.b. this used to also be used for concurrency control, but that is now handled by
+          // shared advisory locks taken using the current sync tick as the id, which are waited on
+          // by an exclusive lock taken prior to starting a snapshot - so this is now purely for
+          // saving with a unique tick
+          const { tock } = await this.tickTockGlobalClock(transactingModels);
+
+          // // run any side effects for each model
+          // // eg: resolving duplicated patient display IDs
+          // await incomingSyncHook(database, modelsToInclude, sessionId);
```

Comment:
TODO? Delete?

Reply:
Probably deleting, haven't found any use cases for that yet